assets/shaders/extended_material.wgsl

```diff
@@ -0,0 +1,36 @@
+#import bevy_pbr::mesh_vertex_output    MeshVertexOutput
+#import bevy_pbr::pbr_functions         PbrInput, apply_pbr_lighting
+#import bevy_pbr::pbr_fragment          pbr_input_from_standard_material, in_shader_post_processing
+
+struct MyExtendedMaterial {
+    quantize_steps: u32,
+}
+
+@group(1) @binding(100)
```

Comment:
I don't love telling people to use 100. I feel like we should
* Pick a smaller number (50?). Ideally we could make it a constant and shader def and have people do `$X + 1, $X + 2, etc`
* Give a specific callout on why we're using 100 here (to not overlap with the standard material bindings, which may change between bevy versions). Also add it in the example rust code.

Reply:
> `$X + 1, $X + 2, etc`
this doesn't work unfortunately. we can make a shader def for the offset but wgsl doesn't allow `@binding(100 + 1)`. this will hopefully be resolved at some point by auto bindings, but definitely not for 0.12 (and probably not for 0.13 - i hope to get the view group done for 0.13 as a start).
>Give a specific callout on why we're using 100 here
👍

Reply:
Ah yeah I forgot to mention that we _can't_ do it yet because naga doesn't allow constant math like that yet.

Reply:
if we don't move to auto bindings then yeah, but not as part of this pr ...

Reply:
also just to note an additional complexity - extensions can be chained / applied on top of one another so the user base isn't foolproof either.